blockly-core/core/block_space_editor.js

```diff
@@ -233,8 +238,25 @@ Blockly.BlockSpaceEditor.prototype.addFlyout_ = function() {
   flyout.autoClose = false;
   // Insert the flyout behind the blockSpace so that blocks appear on top.
   goog.dom.insertSiblingBefore(flyoutSvg, this.blockSpace.svgGroup_);
+};
+
+/**
+ * Gather and return a list of delete rectangles related to this editor.
+ * @public
+ * @return {Array.<Rect>} A list of delete area rectangles.
+ */
+Blockly.BlockSpaceEditor.prototype.getDeleteAreas = function() {
+  var deleteAreas = [];
+
+  if (this.flyout_) {
+    goog.array.extend(deleteAreas, this.flyout_.getRect());
+  }
+
+  if (this.toolbox) {
+    goog.array.extend(deleteAreas, this.toolbox.getRect());
+  }
 
-  this.addChangeListener(this.flyoutBumpOrDeleteOutOfBoundsBlocks_);
```

Comment:
Renamed method and moved hookup to createDom_ in this file, as the bump logic is no longer dependent on a flyout being present.